js/jquery.utils.js

```diff
@@ -433,3 +433,56 @@ function scrollToElement($element, duration) {
         }
     }
 }
+
+
+;(function($, window, document) {
+
+    $.fn.scrollToElement = function($element, duration) {
+        // Scrolls to the top of the given element if it isn't fully visible in the viewport.
+        var $viewport = $(this);
+
+        if ( $viewport.is($(window)) ) {
+            var viewportTop = 0;
+            var viewportBottom = viewportTop + $viewport.height();
+        } else {
+            var viewportTop = $viewport.offset().top;
+            var viewportBottom = viewportTop + $viewport.outerHeight();    
+        }
+        var viewportScrollTop = $viewport.scrollTop();      
+        var elementTop = $element.offset().top;
+        var elementBottom = elementTop + $element.outerHeight();
+        
+        if ( elementBottom > viewportBottom  ) {
+            var newViewportScrollTop = $viewport.scrollTop() + elementBottom - viewportBottom ;            
```

Comment:
add comment
// element below bottom of viewport - scroll down so element is visible